lib/expander/src/Repository.php

```diff
@@ -81,6 +81,34 @@ public function readModule($id, $type) {
         return $model;
     }
 
+
+    /**
+     * Reads module from the store with the given contextinstanceid.
+     * @param String $contextinstanceid
+     * @return PhpObj
+     */
+
+    public function readModuleByContext($contextinstanceid)
+    {
+        $model = $this->readObject($contextinstanceid,'course_modules');
+        return $model;
+    }
+
+    /**
+     * Reads module from the store with the given cm.
+     * @param String $cm
+     * @return PhpObj
+     */
+
+
+    public function readModuleType($cm)
```

Comment:
Oh, c'mon  Ryan  !  It's Friday !  :D     You are right !  It's seems like I overlooked that.  I will change it !